app/src/main/kotlin/com/hippo/ehviewer/Settings.kt

```diff
@@ -217,6 +217,8 @@ object Settings : DataStorePreferences(null) {
     val imageScaleType = intPref("pref_image_scale_type_key", 1)
     val landscapeZoom = boolPref("landscape_zoom", false)
     val zoomStart = intPref("pref_zoom_start_key", 1)
+    val showNavigationOverlayNewUser = boolPref("reader_navigation_overlay_new_user", true)
+    val showNavigationOverlayOnStart = boolPref("reader_navigation_overlay_on_start", false)
```

Comment:
We don't have to keep it in `Settings`
var latestGuidedNavigationType by rememberMutableStateInDataStore("Navigation") { DisabledNavigation }
AnimatedVisibility(
 visible = latestGuidedNavigationType != navigationType,
 modifier = onClick { latestGuidedNavigationType = navigationType}
)

This should be much simpler

Reply:
Addition: let counts TappingInvertMode
Just add a data class
```
@Serialize
data class NavigationAndInvertMode(
val navigation: NavigationType
val invertMode: InvertMode
)

Reply:
We want to show the overlay when the reading mode changes even if the navigation type is the same (e.g. from LTR to RTL).

Reply:
Then take reading mode into that data class?
And rename it to `NavigationGuideInvalidation`

Reply:
The overlay would disappear if navigation mode is changed from A -> B -> A, but we want it to show.

Reply:
> The overlay would disappear if navigation mode is changed from A -> B -> A, but we want it to show.

Let's add a timestamp to navigationType, we reset it to current timestamp when set it explicitly in settings.

Reply:
That doesn't sound like the right solution.
Let's sum it up, we want:
1. Show the overlay whenever reading mode/navigation type/invert mode changes
2. Don't show the overlay on start unless it's forced (for new user or an option, we haven't decided whether to implement the later yet)

Reply:
2 can be concluded to 1(for new user, modes changed from nothing to default)

Reply:
We need to pass navigation regions, which are already tied with reading mode/navigation type/invert mode changes, to `NavigationOverlay` anyway, is it necessary to create a new state?

Reply:
> We need to pass navigation regions, which are already tied with reading mode/navigation type/invert mode changes, to `NavigationOverlay` anyway, is it necessary to create a new state?

Nope, just reuse it I think.

Reply:
I don't think serializing the whole list is a good idea, and I don't think we can as we are relying on instance equality of the regions.

Reply:
> I don't think serializing the whole list is a good idea, and I don't think we can as we are relying on instance equality of the regions.

Then why not combine reading mode/navigation type/invert mode , and make regions a inner cached not Serialized field?

Reply:
Moreover we should not assume reading mode will not change outside reader, as we may allow to configure reader settings in SettingScreen

Reply:
We didn't?

Reply:
> We didn't?

I mean through persist the latest guided combined state, we naturally don't need to flip any flags to allow reshow guide

Reply:
It becomes a little complicated if we want to do that.
1. When we do A -> B -> A loop in reader, we expect the overlay to show again
2. When we do A -> B -> A loop in settings, we expect the overlay not to show again

Reply:
Anyway, I think we follow the old reader's behavior for now, it can be changed in the future.